### PR TITLE
Use msbuild /restore instead of a separate process

### LINKED ISSUE #7009: Cannot pass /clp:NoSummary to dotnet build
## Steps to reproduce

```
dotnet build /clp:NoSummary
```

## Expected  behavior

No build summary shall be displayed (e.g. duplicated errors).

The behaviour should be the same as using it with msbuild:

```
$ dotnet msbuild /clp:NoSummary
Microsoft (R) Build Engine version 15.3.388.41745 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

Program.cs(5,21): error CS0116: A namespace cannot directly contain members such as fields or methods [/Users/martin.ullrich/tmp/tmp.csproj]
```

## Actual behavior

Build summary is displayed:

```
$ dotnet build /clp:NoSummary
Microsoft (R) Build Engine version 15.3.388.41745 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

Program.cs(5,21): error CS0116: A namespace cannot directly contain members such as fields or methods [/Users/martin.ullrich/tmp/tmp.csproj]

Build FAILED.

Program.cs(5,21): error CS0116: A namespace cannot directly contain members such as fields or methods [/Users/martin.ullrich/tmp/tmp.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:01.50
```

This seems to be because the implementation of the `build` verb always appends `/clp:Summary` to the arguments, overriding the passed in arguments.

## Environment data
`dotnet --info` output:

```
.NET Command Line Tools (2.0.0-preview2-006502)

Product Information:
 Version:            2.0.0-preview2-006502
 Commit SHA-1 hash:  07dfded669

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.12
 OS Platform: Darwin
 RID:         osx.10.12-x64
 Base Path:   /usr/local/share/dotnet/sdk/2.0.0-preview2-006502/

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0-preview2-25407-01
  Build    : 40c565230930ead58a50719c0ec799df77bddee9
```

### LINKED ISSUE #7696: Use MSBuild /restore for implicit restore
Commands that perform an implicit restore (invoke MSBuild /t:Restore and then build normally) should move to use the `/restore` command-line flag to reduce JIT compile time.

See https://github.com/Microsoft/msbuild/pull/2414

### PR DESCRIPTION
**Pull request by nguerrera**
Use msbuild /restore instead of separate invocations where possible

Fix #7696

This saves 300ms on my machine for incremental build of a simple console app (from 1.4s to 1.1s).

We still have to miss out on this optimization when there is -f|--framework argument because we cannot force a TargetFramework global property on to the restore evaluation. Doing so completely breaks restore by applying the TargetFramework to all projects transitively. The correct behavior is to restore for all frameworks, then build/publish/etc for the given target framework. Achieving that still requires two distinct msbuild process invocations.

This also changes the verbosity of implicit restore from quiet to that of the subsequent command (default=minimal). Similar to global properties, we cannot specify a distinct console verbosity for the /restore portion of the overall execution. For consistency, we apply the same verbosity change to the case where we still use two separate msbuild invocations.

Output Before:
```
[Logo]
  qwert -> D:\temp\qwert\bin\Debug\netcoreapp2.0\qwert.dll.
[Summary]
```

Output After:
```
[Logo]
  Restoring packages for D:\temp\qwert\qwert.csproj...
  Installing Newtonsoft.Json 10.0.3.
  Restore completed in 763.37 ms for D:\temp\qwert\qwert.csproj.
  qwert -> D:\temp\qwert\bin\Debug\netcoreapp2.0\qwert.dll
[Summary]
```

In the incremental no-op restore case, only the line about "Restore completed in X for Y" is added. 

https://github.com/NuGet/Home/issues/4695 tracks reducing restores output when verbosity is minimal.

This also fixes an issue where the separate restore invocation's msbuild log would be overwritten by the subsequent command execution. However, this remains unfixed in the case where we still use two separate msbuild invocations.

The first two commits are strictly separate, but were getting in the way of test changes I needed to make so I tackled them here.
* Remove necessary handling of verbosity in RestoreCommand
* Fix #7009, Cannot pass /clp:NoSummary to dotnet build

### CODE REVIEW

**Comment by nguerrera** [comment]
@dotnet-bot Test CentOS7.1 x64 Debug Build

**Comment by nguerrera** [comment]
cc @davkean @AndyGerlicher 

@livarcocc Can you take another look. I changed it quite a bit since your review.

File: src/dotnet/commands/dotnet-build/BuildCommand.cs [hunk 1]
```diff
@@ -37,6 +37,8 @@ public static BuildCommand FromArgs(string[] args, string msbuildPath = null)
 
             var appliedBuildOptions = result["dotnet"]["build"];
 
+            msbuildArgs.Add($"/clp:Summary");
```
Comment (livarcocc) [inline src/dotnet/commands/dotnet-build/BuildCommand.cs]: This is so that this can be overwritten?

Comment (nguerrera) [reply src/dotnet/commands/dotnet-build/BuildCommand.cs]: Correct. As it was, dotnet build /clp:NoSummary was broken.

**Pull request merged**